### PR TITLE
Misc: Updated .gitignore

### LINKED ISSUE #5371: misc: update test/mpi/.gitignore
We are supposed to add every test binary in `test/mpi/.gitignore`. We have not been keep it up-to-date. This need be cleaned up.

### PR DESCRIPTION
**Pull request by travisk1154**
## Pull Request Description
Updated .gitignore with new binaries from newly created tests and removed entries for files that no longer existed

Fixes #5371 

## Expected Impact
Developers will now be able to run the new script to automatically add their new test binaries to .gitignore.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: test/mpi/update_gitignore_binaries.py
Comment (hzhou) [inline test/mpi/update_gitignore_binaries.py:8]: We only need to update `.gitignore` every once in a while, so I am not sure it is worth it to maintain a separate script for it. Consider that keeping the `ignored_binaries` always updated is as challenging as keeping `.gitignore` updated itself. I suggest remove this script from the PR.

File: test/mpi/.gitignore
Comment (raffenet) [inline test/mpi/.gitignore:1694]: It would be nice if we could keep these grouped together (alphabetically sorted) with tests from the same directory rather than tacked on at the end of the file.

Comment (hzhou) [reply test/mpi/.gitignore:1694]: I advised him to keep it separate to ease the review. @travisk1154 could you sort them with a 2nd commit?

Comment (travisk1154) [reply test/mpi/.gitignore:1694]: Yea sure thing. Should there also be a separate commit for deletions from .gitignore?

Comment (hzhou) [reply test/mpi/.gitignore:1694]: You can delete the entries in the first commit. It doesn't affect the review. In addition, if you *insert* the new entries rather than reorder the existing ones, you can do all of them in one commit as well.

File: test/mpi/.gitignore
Comment (raffenet) [inline test/mpi/.gitignore:396]: The removed entries here are listed as `EXTRA_PROGRAMS` in `Makefile.am`. I think they are valid, though they are not built by `make all`.

Comment (travisk1154) [reply test/mpi/.gitignore:396]: Ok I'll add them back. That's just the /errhan/ entries?

File: test/mpi/.gitignore
Comment (raffenet) [inline test/mpi/.gitignore:1055]: This is another instance of `EXTRA_PROGRAMS`

File: test/mpi/.gitignore
Comment (raffenet) [inline test/mpi/.gitignore]: these are valid. they are temporary files created by the I/O interface tests.

File: test/mpi/.gitignore
Comment (raffenet) [inline test/mpi/.gitignore:1116]: this looks like its valid. another temporary file created by a test - `manyconnect.in`.

File: test/mpi/.gitignore
Comment (raffenet) [reply test/mpi/.gitignore]: I see `io/testfile.*` is moved to the new block and then moved back in the sorting commit. but `io/test.ord` should not be removed.

File: test/mpi/.gitignore
Comment (raffenet) [inline test/mpi/.gitignore:30]: this one is valid, too. it is a temp file created by `test/mpi/basic/netmpi.c`

**Review by raffenet — COMMENTED** [review]
Did one more pass and found another valid entry removed. I think this is the last issue.

File: test/mpi/.gitignore
Comment (raffenet) [reply test/mpi/.gitignore:30]: it looks like this entry was removed in this commit, and then added back in the sorting commit. rather than nitpick on this, can we just squash these 2 commits together before merge?

**Review by raffenet — APPROVED** [review]
I think this is good to go. I updated the PR title since it no longer contains a script. I also think it would be better to squash before merge, but leave that up to you.

**Pull request merged**
